File: frontend/src/api/jobs.ts
import { useQuery } from "@tanstack/react-query";
import { apiRequest } from "./client";

export type JobStatus =
  | "PENDING"
  | "RUNNING"
  | "SUCCESS"
  | "FAILED"
  | "CANCELLED";

export type JobType = "ON_DEMAND" | "SCHEDULED" | "RETRY" | "BACKFILL";

export interface Job {
  job_id: number;
  source_id: number;
  job_type: JobType;
  status: JobStatus;
  requested_by: number | null;
  parameters: Record<string, unknown>;
  started_at: string | null;
  finished_at: string | null;
  input_count: number;
  output_count: number;
  error_count: number;
  error_message: string | null;
  created_at: string;
}

export interface ListJobsParams {
  source_id?: number;
  status?: JobStatus;
  job_type?: JobType;
  from?: string;
  to?: string;
  limit?: number;
  offset?: number;
}

export function useJobs(params: ListJobsParams = {}) {
  return useQuery({
    queryKey: ["jobs", params],
    queryFn: () => apiRequest<Job[]>("/v1/jobs", { params: { ...params } }),
  });
}

export function useJob(jobId: number | null) {
  return useQuery({
    queryKey: ["jobs", jobId],
    enabled: jobId != null,
    queryFn: () => apiRequest<Job>(`/v1/jobs/${jobId}`),
  });
}

File: backend/app/schemas/jobs.py
"""Pydantic DTOs — `run.ingest_job` 조회."""

from __future__ import annotations

from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, ConfigDict, Field

JobStatus = Literal["PENDING", "RUNNING", "SUCCESS", "FAILED", "CANCELLED"]
JobType = Literal["ON_DEMAND", "SCHEDULED", "RETRY", "BACKFILL"]


class JobOut(BaseModel):
    """단일 ingest_job 직렬화."""

    model_config = ConfigDict(from_attributes=True)

    job_id: int
    source_id: int
    job_type: JobType
    status: JobStatus
    requested_by: int | None = None
    parameters: dict[str, Any] = Field(default_factory=dict)
    started_at: datetime | None = None
    finished_at: datetime | None = None
    input_count: int = 0
    output_count: int = 0
    error_count: int = 0
    error_message: str | None = None
    created_at: datetime


__all__ = ["JobOut", "JobStatus", "JobType"]
